File: Binary Tree Postorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<TreeNode*> arr = {root};
        deque<int> ans;
        while (!arr.empty()) {
            auto node = arr.back();
            arr.pop_back();
            if (node != nullptr) {
                ans.push_front(node->val);
                arr.push_back(node->left);
                arr.push_back(node->right);
            }
        }
        vector<int> a(ans.begin(), ans.end());
        return a;
    }
};

File: Binary Tree Postorder Traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
    arr := []*TreeNode{root}
    ans := make([]int, 0)
    for len(arr) > 0 {
        node := arr[len(arr)-1]
        arr = arr[:len(arr)-1]
        if node != nil {
            ans = append(ans, node.Val)
            arr = append(arr, node.Left)
            arr = append(arr, node.Right)
        }
    }
    n := len(ans) - 1
    i := 0
    for i < n {
        ans[i], ans[n] = ans[n], ans[i]
        i++
        n--
    }
    return ans
}
